### PR TITLE
Handle parse failure for setting spec constants

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: tools/opt/opt.cpp [hunk 1]
```diff
@@ -103,11 +103,18 @@ int main(int argc, char** argv) {
...
+                    "--set-spec-const-default-value: %s\n",
+                    argv[argi]);
+            return 1;
+          }
           pass_manager.AddPass<opt::SetSpecConstantDefaultValuePass>(
               std::move(*spec_ids_vals));
         } else {
           fprintf(stderr,
-                  "error: Expect a string of <spec id>-<default value> pairs.");
+                  "error: Expected a string of <spec id>:<default value> pairs.");
```
Comment (antiagainst) [inline tools/opt/opt.cpp]: It seems we'd better also to change the representation in the help message to `<spec id>:<default value>`.

File: tools/opt/opt.cpp [hunk 2]
```diff
@@ -103,11 +103,18 @@ int main(int argc, char** argv) {
           auto spec_ids_vals =
               opt::SetSpecConstantDefaultValuePass::ParseDefaultValuesString(
                   argv[argi]);
+          if (!spec_ids_vals) {
```
Comment (antiagainst) [inline tools/opt/opt.cpp]: Interesting! Forgot the case that `ParseDefaultValuesString` can also return `nullptr`. Thanks for fixing it!

File: tools/opt/opt.cpp (hunk 1 above)
Comment (dneto0) [inline tools/opt/opt.cpp]: Fixed in next commit

**Comment by dneto0** [comment]
Keeping the approval.

**Pull request merged**